src/main/kotlin/com/kotlindiscord/bot/extensions/FilterExtension.kt

```diff
@@ -0,0 +1,113 @@
+package com.kotlindiscord.bot.extensions
+
+import com.gitlab.kordlib.core.event.message.MessageCreateEvent
+import com.gitlab.kordlib.core.event.message.MessageUpdateEvent
+import com.kotlindiscord.bot.config.config
+import com.kotlindiscord.bot.defaultCheck
+import com.kotlindiscord.bot.enums.Roles
+import com.kotlindiscord.bot.filtering.*
+import com.kotlindiscord.kord.extensions.ExtensibleBot
+import com.kotlindiscord.kord.extensions.checks.topRoleLower
+import com.kotlindiscord.kord.extensions.extensions.Extension
+import mu.KotlinLogging
+
+private val logger = KotlinLogging.logger {}
+private val spoilerRegex = Regex("""(\|\|)(.*?)(\|\|)""", RegexOption.DOT_MATCHES_ALL)
+
+/**
+ * Filter extension, for filtering messages and alerting staff based on message content.
+ *
+ * This is a relatively complex extension, which is concerned with a lot of potential
+ * edge cases and tricky filtering. In order to lower the cognitive load, filtering
+ * behaviour is split up into a bunch of [Filter] classes that work independently
+ * of the rest of this system.
+ */
+class FilterExtension(bot: ExtensibleBot) : Extension(bot) {
+    override val name: String = "filter"
+
+    private val filters: Array<Filter> = arrayOf(
+        InviteFilter(bot),
+        AttachmentFilter(bot),
+        DomainFilter(bot),
+
+        // Non-actioning filters come last, in case a message was already removed.
+        EmbedFilter(bot),
+        RegexFilter(bot)
+    )
+
+    /**
+     * Sanitize message content for filtering.
+     *
+     * This currently removes the pipes from spoilered text.
+     *
+     * @param content Message content to sanitize.
+     * @return Sanitized message content.
+     */
+    fun sanitizeMessage(content: String): String = spoilerRegex.replace(content, "$2")
```

Comment:
I've been thinking about this - and I do remove backslashes in a couple filters - but at least in some cases I feel like it might mess with things such as domains that contain other domains. I'm not sure.

I wonder why PyDis didn't do it.